test/expected/alter.out

```diff
@@ -734,6 +734,34 @@ SELECT hypertable_constraint_name, constraint_name from _timescaledb_catalog.chu
 
 INSERT INTO t_hypertable AS h VALUES ( 1, '2020-01-01 00:01:00', 3.2) ON CONFLICT (id, time) DO UPDATE SET value = h.value + EXCLUDED.value;
 ROLLBACK;
+-- predicate reconstruction when attnos are different in hypertable and chunk
+CREATE TABLE p_hypertable (a integer not null, b integer, c integer);
+SELECT create_hypertable('p_hypertable', 'a', chunk_time_interval => int '3');
+     create_hypertable      
+----------------------------
+ (14,public,p_hypertable,t)
+(1 row)
+
+BEGIN;
+ALTER TABLE p_hypertable DROP COLUMN b, ADD COLUMN d boolean;
+CREATE INDEX idx_ht ON p_hypertable(a, c) WHERE d = FALSE;
+END;
+INSERT INTO p_hypertable(a, c, d) VALUES (1, 1, FALSE);
+\d+ _timescaledb_internal._hyper_14_28_chunk
```

Comment:
Isn't a consensus to use `test/sql/utils/testsupport.sql` functions to show metadata instead of psql commands?? I.e.: `test.show_columns`, `test.show_indexes`

Reply:
That's good to know, `\d+` is still used in some tests, e.g. when displaying view queries.

Reply:
hardcoded chunk names should be avoided too if possible

Reply:
Hmmm. We need to actually describe the chunk to ensure that it's partial index shows up properly here.

Reply:
> Isn't a consensus to use `test/sql/utils/testsupport.sql` functions to show metadata instead of psql commands?? I.e.: `test.show_columns`, `test.show_indexes`

For this test case we require a path which reconstructs the predicate to include deparsing of column 'd'. As per my understanding suggested functions are not covering this.

Reply:
AIUI, `test.show_indexes` function has an `Expr` column output which should show the predicate expression.

Reply:
> AIUI, `test.show_indexes` function has an `Expr` column output which should show the predicate expression.

```SELECT * FROM test.show_indexes('_timescaledb_internal._hyper_14_28_chunk');
                            Index                            | Columns | Expr | Unique | Primary | Exclusion | Tablespace 
-------------------------------------------------------------+---------+------+--------+---------+-----------+------------
 _timescaledb_internal._hyper_14_28_chunk_idx_ht             | {a,c}   |      | f      | f       | f         | 
 _timescaledb_internal._hyper_14_28_chunk_p_hypertable_a_idx | {a}     |      | f      | f       | f         | 
(2 rows)```
Doesn't give any information of column 'd' here.

Reply:
Ah, ok. Can stick to `\d+` for now in that case.

Reply:
@RafiaSabih @nikkhils sorry I missed your comments here, anyway just FYI there is a `test.show_constraints` there:

```sql
select * from test.show_constraints('_timescaledb_internal._hyper_1_1_chunk'::regclass);
  Constraint  | Type | Columns | Index |          Expr          | Deferrable | Deferred | Validated 
--------------+------+---------+-------+------------------------+------------+----------+-----------
 constraint_1 | c    | {a}     | -     | ((a >= 0) AND (a < 3)) | f          | f        | t
(1 row)
```